rmf_task/include/rmf_task/requests/ChargeBattery.hpp

```diff
@@ -27,64 +28,46 @@
 #include <rmf_battery/MotionPowerSink.hpp>
 #include <rmf_battery/DevicePowerSink.hpp>
 
-#include <rmf_utils/optional.hpp>
-
 #include <rmf_task/agv/State.hpp>
 #include <rmf_task/Request.hpp>
 #include <rmf_task/Estimate.hpp>
 
 namespace rmf_task {
 namespace requests {
 
-class ChargeBatteryDescription : public rmf_task::Request::Description
+//==============================================================================
+class ChargeBattery
 {
 public:
 
-  static DescriptionPtr make(
-    rmf_battery::agv::BatterySystem battery_system,
-    rmf_battery::ConstMotionPowerSinkPtr motion_sink,
-    rmf_battery::ConstDevicePowerSinkPtr device_sink,
-    std::shared_ptr<const rmf_traffic::agv::Planner> planner,
-    rmf_traffic::Time start_time,
-    bool drain_battery = true);
+  class Model;
 
-  rmf_utils::optional<rmf_task::Estimate> estimate_finish(
-    const agv::State& initial_state,
-    const agv::Constraints& task_planning_constraints,
-    const std::shared_ptr<EstimateCache> estimate_cache) const final;
+  class Description : public Request::Description
+  {
+  public:
 
-  rmf_traffic::Duration invariant_duration() const final;
+    static DescriptionPtr make(
+      double max_charge_soc = 1.0);
```

Comment:
Good point. Removed `max_charge_soc` here. The `recharge_soc` will not be available via the `parameters` as it is in `Constraints`. But since we need this value only inside `estimate_finish`, we can get it from `task_planning_constraints` that is passed along.
https://github.com/open-rmf/rmf_task/pull/15/commits/4c36e84e6709e3d765fa357850b13eb1f62aa955